discovery/spec.md

```diff
@@ -964,61 +980,51 @@ Content-Type: application/json
 
 {
   "id": "{id}",
-  "epoch": "{id}",
+  "epoch": "{epoch}",
   ... remainder of Service attributes ...
 }
 ```
 
-#### `DELETE /services/{id}`
+#### `DELETE /services/{id}[?epoch={epoch}]`
 
-This MUST delete the Service at the referenced URL.
+This MUST delete the Service as indetified by the `id` in the URL. If there
+is no Service with the specified `id` then it is considered to have already
+been deleted and the request will have no change to the Discovery Endpoint.
+A missing Service is not considered an error condition.
 
-The request MAY include a body. Any such body MUST be an object containing an
-`id` matching the `{id}` in the path. That object MAY include an `epoch`
-value. If the Service corresponding with the specified `id` has a higher
-recorded `epoch` value than given, the request MUST fail.
+The request URL MAY include an OPTIONAL `epoch` query parameter, and if
+present and is not greater than the Service's current `epoch` value, then the
+request MUST fail.
```

Comment:
hmmm but there's no body defined for this operation. Added a sentence to make that clear.

Reply:
There used to be and I would have thought there still would be.  This wouldn't block use but would constrain the DX to the curl side rather than code side.

Reply:
Can you elaborate? Are you saying that not having a Body is problematic? Right now the spec allows for a Body but we basically ignore it as it's totally unneeded.

Reply:
As currently written, the body is potentially the exclusive source of the epoch.

Reply:
Not to ignore that you want to support curl and the like.  That seems reasonable as it gives people writing curl statements a better experience.

If you're writing code and you've already written code against the POST endpoint then you have the code to write the JSON into the body and could reuse that.  Often that feels like the more appropriate way to do things: "here, here's the thing I want deleted".

Reply:
My request was only syntactic sugar.  I think the code reuse goes differently than described but I'm okay with not supporting a body.  It's simpler.  If we allow it I could see people trying to supply an epoch there (and not in the URL params) so ignoring it comes at some potential impact.